python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -90,6 +91,24 @@ def get_target_root_asset_partitions(self) -> Iterable[AssetKeyPartitionKey]:
             self.target_subset.filter_asset_keys(root_asset_keys).iterate_asset_partitions()
         )
 
+    def get_target_root_partitions_subset(self) -> Optional[PartitionsSubset]:
+        root_asset_keys = (
+            AssetSelection.keys(*self.target_subset.asset_keys)
+            .sources()
+            .resolve(self.target_subset.asset_graph)
+        )
+
+        if (
+            not root_asset_keys
+            or self.target_subset.asset_graph.get_partitions_def(next(iter(root_asset_keys)))
```

Comment:
I don't think this is allowed. I guess it could happen if someone removes the partitions defs from an asset that was previously partitioned?